yoti-stub/src/services/YotiRequestProcessor.ts

```diff
@@ -61,1033 +60,1126 @@ import {GET_INSTRUCTIONS_PDF_404} from "../data/getInstructionsPdf/getInstructio
+            this.logger.info("Yoti Session Item", JSON.stringify(yotiSessionItem));
+            return new Response(HttpCodesEnum.CREATED, JSON.stringify(yotiSessionItem));
+        }
+
+        if (lastFullNameChars[0] === '3' || lastFullNameChars[0] === '2') {
+            return new Response(HttpCodesEnum.CREATED, JSON.stringify(yotiSessionItem));
+        }
+
+        switch (lastFullNameChars) {
+            case '1400':
+                this.logger.info({message: "last 4 ID chars", lastFullNameChars});
+                return new Response(HttpCodesEnum.BAD_REQUEST, JSON.stringify(POST_SESSIONS_400))
+            case '1401':
+                this.logger.info({message: "last 4 ID chars", lastFullNameChars});
+                return new Response(HttpCodesEnum.UNAUTHORIZED, JSON.stringify(POST_SESSIONS_401))
+            case '1403':
+                this.logger.info({message: "last 4 ID chars", lastFullNameChars});
+                return new Response(HttpCodesEnum.FORBIDDEN, JSON.stringify(POST_SESSIONS_403))
+            case '1404':
+                this.logger.info({message: "last 4 ID chars", lastFullNameChars});
+                return new Response(HttpCodesEnum.NOT_FOUND, JSON.stringify(POST_SESSIONS_404))
+            case '1503':
+                this.logger.info({message: "last 4 ID chars", lastFullNameChars});
+                return new Response(HttpCodesEnum.SERVICE_UNAVAILABLE, JSON.stringify(POST_SESSIONS_503))
+            case '1999':
+                // This will result in 504 timeout currently as sleep interval is 30s
+                this.logger.info("sleeping for 30 secs");
+                await sleep(30000)
+                this.logger.info("I am awake, returning now");
+                return new Response(HttpCodesEnum.CREATED, JSON.stringify(yotiSessionItem));
+            default:
+                return new Response(HttpCodesEnum.SERVER_ERROR, `Incoming user_tracking_id ${yotiSessionId} didn't match any of the use cases`);
+        }
+    }
+
+    /***
+     * GET /sessions/{id}
+     * @param sessionId
+     */
+    async getSession(sessionId: string): Promise<Response> {
+        const lastUuidChars = sessionId.slice(-4);
+        const firstTwoChars = lastUuidChars.slice(0, 2);
+        this.logger.info({message: "last 4 ID chars", lastUuidChars});
+        let modifiedPayload;
+
+        const processPositiveScenario = (lastUuidChars: string, sessionId: string): Response | undefined => {
+            const logger = this.logger;
+            const yotiSessionRequest = new YotiSessionRequest(sessionId);
+
+            if (firstTwoChars === DocumentMapping.UK_DL) { // UK - Driving Licence Scenarios
+                switch (lastUuidChars) {
+                    case '0000': // UK Driving License Success - Face Match automated
+                        logger.debug(JSON.stringify(yotiSessionRequest));
+                        const VALID_DL_RESPONSE_0000 = JSON.parse(JSON.stringify(VALID_DL_RESPONSE));
```

Comment:
would this be better done with a deep copy? see [https://stackoverflow.com/questions/39968366/how-to-deep-copy-a-custom-object-in-javascript](https://stackoverflow.com/questions/39968366/how-to-deep-copy-a-custom-object-in-javascript)